autoextract_poet/page_inputs.py

```diff
@@ -0,0 +1,59 @@
+from typing import ClassVar, List, Optional, Type
+
+import attr
+
+from autoextract_poet.items import (
+    Article,
+    Item,
+    Product,
+)
+
+
+@attr.s(auto_attribs=True)
+class ResponseData:
+    """Container for AutoExtract response data.
+
+    Should not be used directly by providers.
+    Use derived classes like ArticleResponseData and similar.
+
+    API responses are wrapped in a JSON array
+    (this is to facilitate query batching).
+
+    https://doc.scrapinghub.com/autoextract.html#responses
+    """
+
+    item_class: ClassVar[Type[Item]]
+    item_key: ClassVar[str]
+
+    data: dict
+
+    def to_items(self) -> Optional[List[Item]]:
+        return self.item_class.from_list(
```

Comment:
Batch request may contain data of different data types, e.g. some records can be articles, and some can be products. What do you thiink about making a single record an input, not a list? I.e. don't put raw AutoExtract response to self.data, but require a single entry from the list.

Reply:
Agree. Here it should be just a single entry, no list.

Reply:
Method was renamed to singular (`to_item`) and we're using `from_dict` instead of `from_list` now as suggested.